packages/auth0_api_python/src/auth0_api_python/api_client.py

```diff
@@ -14,46 +29,210 @@ class ApiClient:
+            except Exception:
+                raise self._prepare_error(InvalidDpopProofError("Failed to verify DPoP proof"), auth_scheme=scheme)
+
+            if not http_method or not http_url:
+                raise self._prepare_error(
+                    InvalidDpopProofError("Operation indicated DPoP use but the request has no http_method or http_url"), auth_scheme=scheme
+                )
+
+            try:
+                access_token_claims = await self.verify_access_token(token)
+            except VerifyAccessTokenError as e:
+                raise self._prepare_error(e, auth_scheme=scheme)
+
+            cnf_claim = access_token_claims.get("cnf")
+
+            if not cnf_claim:
+                raise self._prepare_error(
+                    InvalidDpopProofError("Operation indicated DPoP use but the JWT Access Token has no jkt confirmation claim"),
+                    auth_scheme=scheme
+                )
+
+            if not isinstance(cnf_claim, dict):
+                raise self._prepare_error(
+                    InvalidDpopProofError("Operation indicated DPoP use but the JWT Access Token has invalid confirmation claim format"),
+                    auth_scheme=scheme
+                )
+            try:
+                await self.verify_dpop_proof(
+                    access_token=token,
+                    proof=dpop_proof,
+                    http_method=http_method,
+                    http_url=http_url
+                )
+            except InvalidDpopProofError as e:
+                raise self._prepare_error(e, auth_scheme=scheme)
+
+            # DPoP binding verification
+            jwk_dict = (await get_unverified_header(dpop_proof))["jwk"]
+            actual_jkt = calculate_jwk_thumbprint(jwk_dict)
+            expected_jkt = cnf_claim.get("jkt")
+
+            if not expected_jkt:
+                raise self._prepare_error(
+                    VerifyAccessTokenError("Access token 'cnf' claim missing 'jkt'"),
+                    auth_scheme=scheme
+                )
+
+            if expected_jkt != actual_jkt:
+                raise self._prepare_error(
+                    VerifyAccessTokenError("JWT Access Token confirmation mismatch"),
+                    auth_scheme=scheme
+                )
+
+            return access_token_claims
+
+        if scheme == "bearer":
+            if dpop_proof:
+                if self.options.dpop_enabled:
+                    raise self._prepare_error(
+                        InvalidAuthSchemeError(
+                            "Operation indicated DPoP use but the request's Authorization HTTP Header scheme is not DPoP"
+                        ),
+                        auth_scheme=scheme
+                    )
+
+            try:
+                claims = await self.verify_access_token(token)
+                if claims.get("cnf") and claims["cnf"].get("jkt"):
+                    if self.options.dpop_enabled:
```

Comment:
If the access token is DPoP-bound, but the scheme used is Bearer, shouldn't an error be thrown regardless of DPoP being enabled or not?

See requirement:
> When DPoP is enabled in any mode, the SDK must reject any request that includes a DPoP-bound access token in the Authorization header with an authentication scheme other than DPoP. See [Section 7.2](https://www.rfc-editor.org/rfc/rfc9449#section-7.2) of the DPoP specification for more information. The SDK must reject the request with a 400 Bad Request response including an invalid_request error, as described in [Section 4.1.2.1](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1) of the OAuth 2.0 specification.